src/notation/view/internal/undoredohistorymodel.cpp

```diff
@@ -26,35 +26,108 @@ using namespace mu::notation;
 using namespace muse;
 
 UndoRedoHistoryModel::UndoRedoHistoryModel(QObject* parent)
-    : QObject(parent), Injectable(iocCtxForQmlObject(this))
+    : QAbstractListModel(parent), Injectable(iocCtxForQmlObject(this))
 {
 }
 
-size_t UndoRedoHistoryModel::undoRedoActionCount() const
+void UndoRedoHistoryModel::classBegin()
```

Comment:
Out of curiosity - what's the reason for doing this (and all the `QQmlParserStatus` stuff) instead of calling an `init` function from a `Component.onCompleted` signal as we do for most models in the app?

Reply:
I wanted to try out this approach once. It might be a bit more idiomatic. If it is successful, we can roll it out to other places too.